provider/datagen/tests/make-testdata.rs

```diff
@@ -0,0 +1,144 @@
+use icu_datagen::fs_exporter::*;
+use icu_datagen::prelude::*;
+use icu_provider::datagen::*;
+use icu_provider::prelude::*;
+use std::collections::BTreeSet;
+use std::path::Path;
+
+#[global_allocator]
+static ALLOC: dhat::Alloc = dhat::Alloc;
+
+include!("../../../tools/testdata-scripts/locales.rs.data"); // TODO where should this live?
+
+#[test]
+fn generate_fs_and_verify_zero_copy() {
+    simple_logger::SimpleLogger::new()
+        .env()
+        .with_level(log::LevelFilter::Info)
+        .init()
+        .unwrap();
+
+    let data_root = Path::new(concat!(core::env!("CARGO_MANIFEST_DIR"), "/tests/data/"));
+
+    let source = SourceData::offline()
+        .with_cldr(data_root.join("cldr"), Default::default())
+        .unwrap()
+        .with_icuexport(data_root.join("icuexport"))
+        .unwrap();
+
+    let json_out = Box::new(
+        FilesystemExporter::try_new(Box::new(Json::pretty()), {
+            let mut options = ExporterOptions::default();
+            options.root = data_root.join("json");
+            options.overwrite = OverwriteOption::RemoveAndReplace;
+            options.fingerprint = true;
+            options
+        })
+        .unwrap(),
+    );
+
+    let postcard_out = Box::new(
+        FilesystemExporter::try_new(Box::<Postcard>::default(), {
+            let mut options = ExporterOptions::default();
+            options.root = data_root.join("postcard");
+            options.overwrite = OverwriteOption::RemoveAndReplace;
+            options.fingerprint = true;
+            options
+        })
+        .unwrap(),
+    );
+
+    let mut options = options::Options::default();
+    options.locales = options::LocaleInclude::Explicit(LOCALES.iter().cloned().collect());
+
+    DatagenProvider::try_new(options, source)
+        .unwrap()
+        .export(
+            icu_datagen::all_keys().into_iter().collect(),
+            MultiExporter::new(vec![json_out, postcard_out]),
+        )
+        .unwrap();
+
+    // don't drop to avoid dhat from printing stats at the end
+    core::mem::forget(dhat::Profiler::new_heap());
+
+    // violations for net_bytes_allocated
+    let mut net_violations = BTreeSet::new();
+    // violations for total_bytes_allocated (but not net_bytes_allocated)
+    let mut total_violations = BTreeSet::new();
+
+    for entry in glob::glob(
+        &data_root
+            .join("postcard/**/*.postcard")
+            .display()
+            .to_string(),
+    )
+    .unwrap()
+    {
+        let entry = entry.unwrap();
+
+        let payload =
+            DataPayload::from_owned_buffer(std::fs::read(&entry).unwrap().into_boxed_slice());
+
+        let key = icu_datagen::key(
+            &entry
+                .strip_prefix(data_root.join("postcard"))
+                .unwrap()
+                .parent()
+                .unwrap()
+                .display()
+                .to_string()
+                .replace('\\', "/"),
+        )
+        .unwrap();
```

Comment:
Issue: Not very pretty. Might cause conflicts with auxiliary keys. Can we avoid this parsing of file paths to keys? Why can't we use `icu_datagen::all_keys()`?

Reply:
Fixed